tests/test_health.py

```diff
@@ -949,3 +949,39 @@ def test_positive_available_space_postgresql12(ansible_module):
 
     :CaseImportance: High
     """
+
+
+def test_positive_check_duplicate_permissions(ansible_module, setup_duplicate_permissions):
+    """Verify duplicate-permissions check
+
+    :id: 192a5943-43cc-4f89-9949-893c8011831a
+
+    :setup:
+        1. foreman-maintain should be installed.
+        2. Setup duplicate permissions in the database
+
```

Comment:
@shubhamsg199 this test doesn't seem to follow all steps from mentioned BZ, could you check if we could cover all? 
```
Test steps:
- Created duplicate permissions.
- Create a role. 
- Create filters and add duplicate permissions to them.
- Run pre-upgrade check "foreman-maintain health check --label duplicate_permissions"```

Reply:
@shubhamsg199 any update on this?

Reply:
@Gauravtalreja1 It seems we cannot add duplicate permissions to a filter. I've tried the above rest steps it gives `permission invalid` error.
Also confirmed this via UI which also has same behaviour.

Looking at the BZ, I think the main purpose is covered i.e duplicate permissions are removed via check, so should be fine.

Reply:
I agree with @shubhamsg199 here. The outcome of the BZ was the addition of the `foreman-maintain` check for duplicate permissions. This check will report a failure if duplicate permissions are present in the `permissions` table of the Foreman DB [1], so I don't think we gain anything in this test by adding the duplicate permissions to a filter and adding the filter to a role.

[1] https://github.com/theforeman/foreman_maintain/blob/16ad4954605928da2cd3e15b1c21513cfb224121/definitions/checks/foreman/check_duplicate_permission.rb#L26

Reply:
@synkd Ack, Thanks for your inputs here!
I think there might be some improvement in the latest SAT filters creation to avoid adding invalid/duplicate filters but couldn't find anything in the Katello project.